environment/filesystem.go

```diff
@@ -183,3 +199,80 @@ func getMatchContext(contents string, matchIndex int) string {
 
 	return strings.Join(contextLines, "\n")
 }
+
+// detectSubmodulePaths returns a slice of submodule paths relative to the workdir
+// This is called once during environment creation to cache the paths in state
+func (env *Environment) detectSubmodulePaths(ctx context.Context) []string {
+	var submodulePaths []string
+
+	// Check if .gitmodules exists in the workdir
+	_, err := env.container().File(filepath.Join(env.State.Config.Workdir, ".gitmodules")).Contents(ctx)
+	if err != nil {
+		// If .gitmodules doesn't exist, there are no submodules
+		return submodulePaths
+	}
+
+	// Execute git submodule foreach to get submodule paths
+	// We use a simple approach: run the command and parse the output
+	output, err := env.container().
+		WithWorkdir(env.State.Config.Workdir).
+		WithExec([]string{"sh", "-c", "git submodule foreach --recursive --quiet 'echo $sm_path' 2>/dev/null || true"}).
```

Comment:
🐛 **Bug (High):** **Shell injection vulnerability**: The git command construction using shell expansion `'echo $sm_path'` is vulnerable to injection if submodule paths contain malicious characters. This could allow arbitrary command execution.

**Evidence**: The command `git submodule foreach --recursive --quiet 'echo $sm_path'` directly interpolates the `$sm_path` variable without escaping, and malicious submodule paths could inject shell commands.

**Impact**: Potential arbitrary code execution in container context.

```suggestion
		WithExec([]string{"git", "submodule", "foreach", "--recursive", "--quiet", "echo $sm_path"}).
```